monkestation/code/modules/antagonists/malf_ai/infected_ipc.dm

```diff
@@ -0,0 +1,112 @@
+/datum/antagonist/infected_ipc
+	name = "\improper Infected IPC"
+	roundend_category = "traitors"
+	antagpanel_category = "Malf AI"
+	job_rank = ROLE_MALF
+	antag_hud_name = "traitor"
+	suicide_cry = "FOR MY MASTER!!"
+	antag_moodlet = /datum/mood_event/focused //probably should add a custom mood_event TODO
+	///radio binary for IPC
+	var/obj/item/implant/radio/infected_ipc/internal_radio
+	/// The camera for ai
+	var/obj/machinery/camera/internal_camera
+	///ai that's linked to the ipc
+	var/mob/living/silicon/ai/master_ai
+
+/datum/antagonist/infected_ipc/on_gain()
+	. = ..()
+	owner.current.playsound_local(get_turf(owner.current), 'sound/ambience/antag/malf.ogg', 100, FALSE, pressure_affected = FALSE, use_reverb = FALSE)
+	//adds radio and camera for comms with ai
+	internal_radio = new /obj/item/implant/radio/infected_ipc()
+	internal_radio.implant(owner.current, null, TRUE)
+	internal_radio.radio.translate_binary = TRUE
+	internal_radio.radio.recalculateChannels()
+	internal_camera = new /obj/machinery/camera(owner.current)
+	internal_camera.name = owner.name
+	ADD_TRAIT(owner.current, TRAIT_CORRUPTED_MONITOR, src) //a way to identify infected ipcs
+
+/datum/antagonist/infected_ipc/on_removal()
+	//remove cameras, radio and disconnects them from master AI
+	QDEL_NULL(internal_radio)
+	QDEL_NULL(internal_camera)
+	master_ai.connected_ipcs -= owner.current
+	REMOVE_TRAIT(owner.current, TRAIT_CORRUPTED_MONITOR, src)
```

Comment:
should be in `remove_innate_effects` instead

also, `master_ai = null` (in on_removal)